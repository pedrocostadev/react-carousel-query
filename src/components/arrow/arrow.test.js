import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Arrow from './arrow';

describe('<Arrow />', () => {
  const onClick = jest.fn();

  test('It should render', () => {
    const button = render(<Arrow variant="left" onClick={onClick} />);
    expect(button).toMatchSnapshot();
  });

  test('It should render using custom render function', () => {
    const getCustomArrowId = (variant) => `${variant}-custom-arrow`;
    const renderArrow = ({ variant }) => (
      <p data-testid={getCustomArrowId(variant)}>{variant}</p>
    );

    const { container, getByTestId } = render(
      <Arrow variant="left" onClick={onClick} renderArrow={renderArrow} />,
    );
    expect(getByTestId(getCustomArrowId('left'))).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('It should call onClick', () => {
    const { getByTestId } = render(
      <Arrow data-testid="arrow-left" variant="left" onClick={onClick} />,
    );
    const arrowButton = getByTestId('arrow-left');
    fireEvent.click(arrowButton);
    expect(onClick).toBeCalledTimes(1);
  });

  test('It should have LEFT classe when variant is left', () => {
    const { getByTestId } = render(
      <Arrow data-testid="arrow-left" variant="left" onClick={onClick} />,
    );
    const arrowButton = getByTestId('arrow-left');
    expect(arrowButton.classList.contains('left')).toBe(true);
  });

  test('It should have RIGHT classe when variant is right', () => {
    const { getByTestId } = render(
      <Arrow data-testid="arrowRight" variant="right" onClick={onClick} />,
    );
    const arrowButton = getByTestId('arrowRight');
    expect(arrowButton.classList.contains('right')).toBe(true);
  });
});
