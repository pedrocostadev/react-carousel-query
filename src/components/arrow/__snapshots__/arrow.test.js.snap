// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Arrow /> It should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="arrowButton left"
        data-testid="arrow-button-left"
      >
        <svg
          fill="none"
          fill-rule="evenodd"
          height="15"
          preserveAspectRatio="xMidYMax meet"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          viewBox="-.2 -.3 8 5"
          width="40"
        >
          <polyline
            points="1 1 3.77984472 4 6.53846154 1"
          />
        </svg>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="arrowButton left"
      data-testid="arrow-button-left"
    >
      <svg
        fill="none"
        fill-rule="evenodd"
        height="15"
        preserveAspectRatio="xMidYMax meet"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        viewBox="-.2 -.3 8 5"
        width="40"
      >
        <polyline
          points="1 1 3.77984472 4 6.53846154 1"
        />
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Arrow /> It should render using custom render function 1`] = `
<div>
  <p
    data-testid="left-custom-arrow"
  >
    left
  </p>
</div>
`;
